src/core/services/slot-refresher.ts

```
@@ -24,35 +31,123 @@ async function isUAP(): Promise<boolean> {
 	}
 
-	refreshSlot(adSlot: AdSlot) {
-		if (!this.config.slots.includes(adSlot.getSlotName())) return;
+	isSlotRefreshable(slotName: string) {
+		return this.config.slots.includes(slotName);
+	}
+
+	/**
+	 * Removes adSlot sizes taller than the first rendered adSlot with the same slotName.
+	 * @param {AdSlot} adSlot - The ad slot to adjust.
+	 */
+	removeHigherSlotSizes(adSlot: AdSlot) {
+		const slotName = adSlot.getSlotName();
+		const slotHeight = adSlot.getElement().clientHeight;
+
+		const slotRefresherConfig = context.get('slotConfig.slotRefresher.sizes');
+		if (!slotRefresherConfig) return;
+
+		const slotHeightLimit = slotRefresherConfig[slotName][1];
+		if (!slotHeightLimit) return;
+
+		const sizeKey = context.get(`slots.${adSlot.getSlotName()}.sizes`);
+
+		if (sizeKey) {
+			context.set(
+				`slots.${adSlot.getSlotName()}.sizes`,
+				sizeKey[0].sizes.filter((size) => size[1] <= slotHeight && size[1] > 3),
+			);
+		}
+		const defaultSizeKey = context.get(`slots.${adSlot.getSlotName()}.defaultSizes`);
+
+		if (defaultSizeKey) {
+			context.set(
+				`slots.${adSlot.getSlotName()}.defaultSizes`,
+				defaultSizeKey.filter((size) => size[1] <= slotHeight && size[1] > 3),
+			);
+		}
+
+		if (adSlot.config.sizes) {
+			adSlot.config.sizes = adSlot.config.sizes.filter((size) => size[1] <= slotHeightLimit);
+		}
 
+		if (adSlot.config.defaultSizes) {
+			adSlot.config.defaultSizes = adSlot.config.defaultSizes.filter(
+				(size) => size[1] <= slotHeightLimit,
+			);
+		}
+
+		const filterCallback = (size: [number, number]) => size[1] <= slotHeightLimit;
+
+		communicationService.emit(eventsRepository.SLOT_REFRESHER_SET_MAXIMUM_SLOT_HEIGHT, {
+			adSlot,
+			filterCallback,
+		});
+	}
+
+	addSlotSizeToContext(adSlot: AdSlot) {
+		const slotWidth = adSlot.getElement().clientWidth;
+		const slotHeight = adSlot.getElement().clientHeight;
+
+		context.set(`slotConfig.slotRefresher.sizes.${adSlot.getSlotName()}`, [slotWidth, slotHeight]);
```

as we do not do anything with this value, except pass them, maybe it can be shorten to:
```suggestion
		context.set(`slotConfig.slotRefresher.sizes.${adSlot.getSlotName()}`, [adSlot.getElement().clientWidth, adSlot.getElement().clientHeight]);
```